### PR TITLE
add karmada operator roadmap

### PR DESCRIPTION
**Pull request by calvin0327**
Hi, before the develoment of the project, we came up with a roadmap, it include core feature about karmada operator. if the roadmap is be determined, we will start follow the roadmap.

### CODE REVIEW

**Comment by calvin0327** [comment]
@lonelyCZ @RainbowMango PTAL

File: Roadmap.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# karmada operator roadmap
+
+The karmada operator is a new project of karmada community designed to manage the lifecycle of karmada. The following is the core feature of the karmada operator that we will work on. It is not a detailed to-do list, but represents overall development direction. It is hoped that more contributors will have a general understanding of project and participate in it.
```
Comment (Poor12) [inline Roadmap.md:3]: all the `karmada` should change to `Karmada`.

File: Roadmap.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+## 2022 H2
+- the `pr` and `issue` template
```
Comment (Poor12) [inline Roadmap.md:6]: I think roadmap is more like a plan for new features about karmada-operator. We should not add some the content of community governance.

File: Roadmap.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+- actions
+  * code lint
+  * build mutil-platform image
+- makefile & scirpts
```
Comment (Poor12) [inline Roadmap.md:10]: Meanwhile, it‘s more user-oriented. I prefer to tell users what you can do with the karmada-operator in the future. Things such as e2e are not needed.

File: Roadmap.md (hunk 1 above)
Comment (calvin0327) [reply Roadmap.md]: updated.

File: Roadmap.md (hunk 2 above)
Comment (calvin0327) [reply Roadmap.md:14]: That's reasonable, but it’s every concise if we only list features of karmada, and the e2e and some CI actions is important part of jobs for us,  so I separate two part content to feature and others. so we can easy focus on features.

Comment (Poor12) [reply Roadmap.md:14]: Maybe we can talk about it on today‘s community meeting.

Comment (calvin0327) [reply Roadmap.md:14]: ok

**Comment by calvin0327** [comment]
@Poor12 On today‘s community meeting, we had discussion on it. There's not a lot of discussion about whether we need to put community governance jobs on the roadmap, do you have any other opinions?

**Comment by Poor12** [comment]
> @Poor12 On today‘s community meeting, we had discussion on it. There's not a lot of discussion about whether we need to put community governance jobs on the roadmap, do you have any other opinions?

I'm ok. cc @RainbowMango for opinions.

**Comment by RainbowMango** [comment]
> There's not a lot of discussion about whether we need to put community governance jobs on the roadmap, do you have any other opinions?

I'm ok with it.

**Comment by calvin0327** [comment]
@RainbowMango thanks, updated

**Comment by RainbowMango** [comment]
I'm really sorry that I didn't say it at the last review. Please update the file name to `ROADMAP.md` with the upper case.
No special reason, just keep consistent with the karmada repo.

**Pull request merged**